### PR TITLE
WIP: Kube watcher to informer

### PR DESCRIPTION
**Pull request by sharadregoti**

### CODE REVIEW

File: gateway/utils/handlers/config.go [hunk 1]
```diff
@@ -225,6 +225,7 @@ func HandleLoadProjects(adminMan *admin.Manager, syncMan *syncman.Manager, confi
 			}
 			syncMan.SetGlobalConfig(c)
 		}
+		syncMan.SetGlobalConfig(c)
```
Comment (YourTechBud) [inline gateway/utils/handlers/config.go]: Why you doing this?

File: gateway/utils/syncman/store_kube.go [hunk 2]
```diff
@@ -46,117 +50,151 @@ func (s *KubeStore) Register() {
...
-					continue
-				}
-				projectMap[v.ID] = v
-
-			case watch.Deleted:
-				configMap := ee.Object.(*v1.ConfigMap)
+		informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				cb(s.getProjects(onAddOrUpdateProjects(obj, projectMap)))
+				logrus.Infof("project added")
```
Comment (YourTechBud) [inline gateway/utils/syncman/store_kube.go]: Make this debug

File: gateway/utils/syncman/store_kube.go [hunk 3]
```diff
@@ -46,117 +50,151 @@ func (s *KubeStore) Register() {
...
-			}
+				cb(s.getProjects(projectMap))
+				logrus.Infof("project deleted")
 
-			cb(s.getProjects(projectMap))
-		}
+			},
+			UpdateFunc: func(old, obj interface{}) {
+				cb(s.getProjects(onAddOrUpdateProjects(obj, projectMap)))
+				logrus.Infof("project updated")
```
Comment (YourTechBud) [inline gateway/utils/syncman/store_kube.go]: Make this debug

**Comment by YourTechBud** [comment]
Your tests are failing

**Comment by YourTechBud** [comment]
You have tested all of this right?

**Comment by sharadregoti** [comment]
yes

**Pull request merged**